azurelinuxagent/common/osutil/redhat.py

```diff
@@ -140,6 +139,30 @@ def get_nm_controlled(self):
 
         return True
 
+    def check_and_recover_nic_state(self, ifname):
+        filepath = "/sys/class/net/{0}/operstate".format(ifname)
+        try:
+            if not os.path.isfile(filepath):
+                logger.warn("Unable to determine primary network interface state, because state file does not exist: "
+                            "{0}".format(filepath))
+            else:
+                content = fileutil.read_file(filepath).rstrip().encode('utf-8')
+                down_states = ["down", "unknown"]
```

Comment:
when do we see 'unknown'?

Reply:
From kernel.org documentation:

> Interface is in unknown state, neither driver nor userspace has set
>  operational state. Interface must be considered for user data as
>  setting operational state has not been implemented in every driver.
 
 If driver isn't setting operation state, a restart of the NM is unlikely to fix this. I think I should remove 'unknown' from this list of states. We can log a warning when interface is in unknown state